hubblestack/extmods/grains/cloud_details.py

```diff
@@ -35,8 +35,13 @@ def _get_aws_details():
     proxies = {'http': None}
 
     try:
+        ttl_header = {'X-aws-ec2-metadata-token-ttl-seconds': '300'}
+        token_url = 'http://169.254.169.254/latest/api/token'
+        token_request = requests.put(token_url, headers=ttl_header)
```

Comment:
You should specify `proxies=proxies` arg as well like we do in `requests.get` call and also specify a timeout similar to .get call. This is required as users might set proxy globally using environment variable. Also timeout helps in safeguarding against corner cases where this URL may not respond.

Reply:
Done.